app/jobs/deposit_job.rb

```diff
@@ -3,16 +3,49 @@
 
 # Deposits a Work into dor-services-app
 class DepositJob < ApplicationJob
+  extend T::Sig
+
   queue_as :default
 
+  sig { params(work: Work).void }
   def perform(work)
-    result = api_client.objects.register(params: create_model(work))
-    work.druid = result.externalIdentifier
+    job_id = deposit(create_model(work))
+    result = nil
+    until result
```

Comment:
Not for this PR, but how would you feel about follow-up work that lets sdr-client manage its async interactions on apps' behalf, like dor-services-client does? https://github.com/sul-dlss/dor-services-client#asynchronous-results That will simplify the DepositJob.

Reply:
Sounds good to me.